Improve release notes for ESP32 support

takasehideki opened the pull request:
This PR tries to include the subminor number of ESP-IDF supported versions because it is sensitive to build firmware for the ESP32 family. Also, v5.2 (v5.2.1) is covered at this time, so this has been added.
Moreover, I added link to target `idf-version` on [esp32-build.yaml](https://github.com/atomvm/AtomVM/blob/main/.github/workflows/esp32-build.yaml#L38). As future development progresses, the information in this table may diverge from the CI build target numbers, but I expect that users can follow this link to the latest status.

This will close #1168

---
These changes are made under both the "Apache 2.0" and the "GNU Lesser General
Public License 2.1 or later" license terms (dual license).

SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later

UncleGrumpy reviewed (commented):
I think to keep the maintenance of this information simpler, it might be best to change the wording to state that we least with the latest release for each of the supported branches, otherwise it get easy to forget to change the specific point releases here when the CI gets updated, and then we have information that is specifically wrong, which could cause even more confusion than the current vagueness of the point release.

takasehideki commented:
I deeply understand your concern, keeping documents fresh is always tough work ;(
So I removed the subminor numbers (rebase my changes) in the table and modified the guidance so that users can refer to the current target versions in `esp32-build.yaml`. I expect that there is no need to update the table and users can catch up on the proper information.

UncleGrumpy commented on doc/release-notes.md.in:
```diff
@@ -64,11 +64,12 @@ AtomVM currently supports the following versions of ESP-IDF:
+| ESP-IDF [v5.2](https://docs.espressif.com/projects/esp-idf/en/v5.2.1/esp32/get-started/index.html) | ✅ |
 
-Building the AtomVM virtual machine for ESP32 is optional.  In most cases, you can simply download a release image from the AtomVM [release](https://github.com/atomvm/AtomVM/releases) repository.  If you wish to work on development of the VM or use one on the additional drivers that are available in the [AtomVM repositories](https://github.com/atomvm) you will to build AtomVM from source.  See th ...
+Building the AtomVM virtual machine for ESP32 is optional.  In most cases, you can simply download a release image from the AtomVM [release](https://github.com/atomvm/AtomVM/releases) repository.  If you wish to work on development of the VM or use one on the additional drivers that are available in the [AtomVM repositories](https://github.com/atomvm) you will to build AtomVM from source.  See th ...
```
I wonder if it might be better to point people to consult a recent workflow run from this link: https://github.com/atomvm/AtomVM/actions/workflows/esp32-build.yaml to check for patch release number that we are testing against, but in general we try to update out CI as soon as new releases are available. Just because any changes to the workflow are likely to change the line number of the version matrix, and it will be hard to remember to update this link when updating the workflow, since it isn't obvious that they are related.

takasehideki replied:
I think so; in fact, at first, I thought that link might be a good idea. But that link you showed does not yet show the version number, and users need to jump to one more link. Or it might be better just to remove the line number `#L38` from the link to the file `esp32-build.yaml`. Or perhaps simply indicate to users that you would use the latest subminor number (patch number).
In summary, which sentence do you think is best to include?

1. Also, please check [esp32-build.yaml](https://github.com/atomvm/AtomVM/actions/workflows/esp32-build.yaml) and the following link for the current versions including the subminor number used in the CI build and its status.
2. Also, please check [`idf-version` on esp32-build.yaml](https://github.com/atomvm/AtomVM/blob/main/.github/workflows/esp32-build.yaml) for the current versions including subminor number used in the CI build.
3. Also, we recommend you to use the latest subminor (patch) versions for source build.

UncleGrumpy replied:
I think 3 is best, but it might be a good idea to include a link to the workflow for the exact version used for testing (as a disclaimer). We try to stay current, but it can take a few days from the time a new IDF version is released before we get an updated workflow merged into the CI.

So maybe, "We recommend you to use the latest subminor (patch) versions for source builds.  You can check the current version used for testing in the [esp32-build.yaml](https://github.com/atomvm/AtomVM/actions/workflows/esp32-build.yaml) workflow."

takasehideki replied:
Thank you for your suggestion! I added that sentence in 709eeb5b

UncleGrumpy reviewed (approved):
This looks good to me 👍
